### PR TITLE
Scripts: Add lint-style script based on stylelint

### PR DESCRIPTION
**Pull request by gziolo**
## Description
This PR adds `lint-style` to `wp-scripts` exposed by `@wordpress/scripts` package.

It helps enforce coding style guidelines for your style files. It uses [stylelint](https://github.com/stylelint/stylelint) with the set of default rules provided. You can override them with your own rules as described in [stylelint user guide](https://github.com/stylelint/stylelint/docs/user-guide.md).

## How has this been tested?
`npm run lint-css` still works properly in the repository.
`./node_modules/.bin/wp-scripts lint-style '**/*.scss'` also works properly

### CODE REVIEW

File: packages/scripts/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 2.5.0 (Unreleased)
+
+### New Feature
+
+- Added support for `lint-style` script ([#XXXXX](https://github.com/WordPress/gutenberg/pull/XXXXX))
```
Comment (adamsilverstein) [inline packages/scripts/CHANGELOG.md]: Assuming we need to add the PR# here?
```suggestion
- Added support for `lint-style` script ([#12722](https://github.com/WordPress/gutenberg/pull/12722))
```

**Review by adamsilverstein — APPROVED** [review]

**Comment by adamsilverstein** [comment]
Looks good to me!

File: packages/scripts/README.md [hunk 2]
```diff
@@ -42,7 +43,7 @@ _Example:_
 ```
 
 This is how you execute the script with presented setup:
-* `npm run lint:js` - lints JavaScripts files in the whole project's.
+* `npm run lint:js` - lints JavaScripts files in the whole project's directory.
```
Comment (ntwb) [inline packages/scripts/README.md]:
```suggestion
* `npm run lint:js` - lints JavaScript files in the entire project's directories.
```

**Comment by ntwb** [comment]
The `.stylelintrc.json → packages/scripts/config/.stylelintrc.json` change copies the Gutenberg specific `.stylelintrc.json` stylelint configuration to be distributed by `@wordpress/scripts` 

This file should remain in its current location and a new file should be created at `packages/scripts/config/.stylelintrc.json` that contains:

```json
{
	"extends": "stylelint-config-wordpress"
}

```

The `README.md` should be updated to:

```md
It uses [stylelint](https://github.com/stylelint/stylelint) with the [stylelint-config-wordpress](https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress) configuration per the [WordPress CSS Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/).
```

**Review by ntwb — CHANGES_REQUESTED** [review]
The notes at https://github.com/WordPress/gutenberg/pull/12722#issuecomment-445585669 should be addressed before merge

**Comment by gziolo** [comment]
Thanks @ntwb for sharing your feedback. I'm in transit and will get back to it later this week. Unless you want to apply proposed changes yourself :)

**Comment by gziolo** [comment]
> I originally wrote the above and what I was going to write here was about using `lint-style` as the command instead of `lint-css`, expanding on this thinking, we would want to add `lint-scss` for the scss config and then have `lint-style` run both `lint-css` and `lint-scss`?

What would be the difference? I assumed they are the same in the context of Gutenberg.

File: packages/scripts/README.md [hunk 3]
```diff
@@ -80,6 +81,23 @@ _Example:_
...
+```json
+{
+	"scripts": {
+		"lint:css": "wp-scripts lint-style '**/*.css'"
+	}
+}
+```
+
+This is how you execute the script with presented setup:
+* `npm run lint:css` - lints CSS files in the whole project's directory.
```
Comment (gziolo) [inline packages/scripts/README.md]:
```suggestion
* `npm run lint:css` - lints CSS files in the entire project's directories.
```

**Comment by gziolo** [comment]
@ntwb I applied all your suggestions in 668eebbfe647892f8d878d23067a2a9a82eb83f4.

**Review by ntwb — APPROVED** [review]

**Review by ntwb — APPROVED** [review]

**Pull request merged**